packages/react-components/src/DocsComponents/Toc.stories.tsx

```diff
@@ -37,26 +69,57 @@ const useTocStyles = makeStyles({
 // };
 
 const navigate = (url: string) => {
-  console.log('Navigate', url);
   addons.getChannel().emit(NAVIGATE_URL, url);
 };
 
 export const nameToHash = (id: string): string => id.toLowerCase().replace(/[^a-z0-9]/gi, '-');
 
-export const Toc = ({ stories }: { stories: PublishedStoreItem[] }) => {
+export const Toc = ({ stories }: { stories: TocItem[] }) => {
+  const [selected, setSelected] = React.useState('');
+
+  const handleObserver = (entries: IntersectionObserverEntry[]) => {
+    for (const entry of entries) {
+      const { intersectionRatio, target } = entry;
+      if (intersectionRatio > 0.5) {
```

Comment:
really nifty feature, I like it 👍

Reply:
+1, if I understood this correctly, we're using the titles as the intersection targets, correct? Does 0.5 make a difference vs 1 or am I missing something?

Reply:
i pulled that ratio from another implementation on the old website. it might depend on what you observe. if its something that COULD be much taller, you don't want to wait for all of it to come into view.